spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -193,6 +337,19 @@ private Schema schemaWithMetadataColumns() {
 
   @Override
   public Scan build() {
+    // if aggregates are pushed down, instead of constructing a SparkBatchQueryScan, creating file
+    // read tasks and sending over the tasks to Spark executors, a SparkLocalScan will be created
+    // and the scan is done locally on the Spark driver instead of the executors. The statistics
+    // info will be retrieved from manifest file and used to build a Spark internal row, which
+    // contains the pushed down aggregate values.
+    if (pushedAggregateRows != null) {
```

Comment:
I think it would be slightly better to create the scan in the aggregation methods. Then this could be `if (localScan != null) { return localScan }` which is a bit more generic.